src/ReachSets/DiscretePost/DecomposedDiscretePost.jl

```diff
@@ -0,0 +1,144 @@
+export DecomposedDiscretePost
+
+"""
+    DecomposedDiscretePost <: DiscretePost
+
+Textbook implementation of a discrete post operator, but with lazy decomposed intersections.
+
+### Fields
+
+- `options` -- an `Options` structure that holds the algorithm-specific options
+
+### Algorithm
+
+The algorithm is based on [Flowpipe-Guard Intersection for Reachability
+Computations with Support Functions](http://spaceex.imag.fr/sites/default/files/frehser_adhs2012.pdf).
+"""
+struct DecomposedDiscretePost <: DiscretePost
+    options::Options
+
+    function DecomposedDiscretePost(𝑂::Options)
+        𝑂copy = copy(𝑂)
+        # TODO: Check why it takes always default value for convex_hull
+        check_aliases_and_add_default_value!(𝑂.dict, 𝑂copy.dict, [:overapproximation], Hyperrectangle)
+        check_aliases_and_add_default_value!(𝑂.dict, 𝑂copy.dict, [:out_vars], Vector{Int}())
+
+        return new(𝑂copy)
+    end
+end
+
+# convenience constructor from pairs of symbols
+DecomposedDiscretePost(𝑂::Pair{Symbol,<:Any}...) = DecomposedDiscretePost(Options(Dict{Symbol,Any}(𝑂)))
+
+# default options for the DecomposedDiscretePost discrete post operator
+DecomposedDiscretePost() = DecomposedDiscretePost(Options())
+
+init(𝒫::DecomposedDiscretePost, 𝒮::AbstractSystem, 𝑂::Options) = init!(𝒫, 𝒮, copy(𝑂))
+
+# TODO: use 𝑂 only?
+function init!(𝒫::DecomposedDiscretePost, 𝒮::AbstractSystem, 𝑂::Options)
+    𝑂[:n] = statedim(𝒮, 1)
+
+    # solver-specific options (adds default values for unspecified options)
+    𝑂out = validate_solver_options_and_add_default_values!(𝑂)
+
+    return 𝑂out
+end
+
+function tube⋂inv!(𝒫::DecomposedDiscretePost,
+                   reach_tube::Vector{<:ReachSet{<:LazySet, N}},
+                   invariant,
+                   Rsets,
+                   start_interval
+                  ) where {N}
+
+    dirs = 𝒫.options[:overapproximation]
+
+    # counts the number of sets R⋂I added to Rsets
+    count = 0
+    @inbounds for reach_set in reach_tube
+        push!(Rsets, ReachSet{LazySet{N}, N}(reach_set.X,
```

Comment:
Add a comment here that the intersection with the invariant has already been computed in the continuous-post operator.